plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryType.java

```diff
@@ -182,6 +194,75 @@ static String bytesToStringConverter(Object value)
         return format("FROM_BASE64('%s')", Base64.getEncoder().encodeToString(slice.getBytes()));
     }
 
+    public static Field toField(String name, Type type)
+    {
+        if (type instanceof ArrayType) {
+            Type elementType = ((ArrayType) type).getElementType();
+            return toInnerFieldBuilder(name, elementType).setMode(REPEATED).build();
+        }
+        return toInnerFieldBuilder(name, type).build();
+    }
+
+    private static Field.Builder toInnerFieldBuilder(String name, Type type)
+    {
+        if (type instanceof RowType) {
+            return Field.newBuilder(name, StandardSQLTypeName.STRUCT, toFieldList((RowType) type));
+        }
+        return Field.newBuilder(name, toStandardSqlTypeName(type));
+    }
```

Comment:
nit: IMO using `Field` as return type, passing `repeated` flag via parameter and using `Builder` only internally makes code a bit easier to read. 
WDYT:
```suggestion
    public static Field toField(String name, Type type)
    {
        if (type instanceof ArrayType) {
            Type elementType = ((ArrayType) type).getElementType();
            return toInnerField(name, elementType, true);
        }
        return toInnerField(name, type, false);
    }

    private static Field toInnerField(String name, Type type, boolean repeated)
    {
        Field.Builder builder;
        if (type instanceof RowType) {
            builder = Field.newBuilder(name, StandardSQLTypeName.STRUCT, toFieldList((RowType) type));
        }
        else {
            builder = Field.newBuilder(name, toStandardSqlTypeName(type));
        }
        if (repeated) {
            builder = builder.setMode(REPEATED);
        }
        return builder.build();
    }
```